presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -170,6 +176,29 @@ public void testTimestamps()
         assertQueryUsingH2Cte("SELECT commit_timestamp, ship_timestamp, orderkey FROM lineitem_ex WHERE year(commit_timestamp) > 1993 and year(ship_timestamp) > 1993 and year(ship_timestamp) - year(commit_timestamp) = 1");
     }
 
+    @Test
+    public void testFloats()
+    {
+        assertQueryUsingH2Cte("SELECT discount_real, tax_real FROM lineitem_ex");
+
+        assertFilterProject("tax_real IS NOT NULL", "count(*)");
+
+        assertFilterProject("tax_real IS NULL", "count(*)");
+
+        assertFilterProject("tax_real > 0.1", "count(*)");
+
+        assertFilterProject("tax_real < 0.03", "discount_real, tax_real");
+
+        assertFilterProject("tax_real < 0.05  AND discount_real > 0.05", "discount_real");
+
+        assertFilterProject("tax_real = discount_real", "discount_real");
+
+        assertFilterProject("discount_real > 0.01 AND tax_real > 0.01 AND (discount_real + tax_real) < 0.08", "discount_real");
+
+        //enable this after adding support for floats in list filter
```

Comment:
A better way would be to do one of the following:

- add a check to Float reader's constructor that throws an error if filter is non-deterministic;
- add support for non-deterministic filters by changing L202 in readerWithFilter to:

```
            outputPositionCount -= filter.getPrecedingPositionsToFail();

            streamPosition++;

            int succeedingPositionsToFail = filter.getSucceedingPositionsToFail();
            if (succeedingPositionsToFail > 0) {
                int positionsToSkip = 0;
                for (int j = 0; j < succeedingPositionsToFail; j++) {
                    i++;
                    int nextPosition = positions[i];
                    positionsToSkip += 1 + nextPosition - streamPosition;
                    streamPosition = nextPosition + 1;
                }
                skip(positionsToSkip);
            }
```

Reply:
Yes, thats right I had this in my local, since this should go into all scalars, wanted to do that in a new commit. Will do the option 1 now, add a check in the constructor